test/test_one_dimensional_shallow_water_model.jl

```diff
@@ -0,0 +1,48 @@
+# # One dimensional shallow water example
+
+using Oceananigans, Oceananigans.Advection
+
+include("../src/Models/ShallowWaterModels/ShallowWaterModels.jl")
```

Comment:
Do you have to use the `include` statements? Might be cleaner to set up the tests like a user would run them until the `ShallowWaterModel` is exported by the top-level `Oceananigans` module, e.g.

```julia
using Oceananigans.Models # should import the shallow water model stuff
```

Reply:
I totally agree adn will make those changes everywhere in this script. Thanks!

Reply:
Unfortuantely, when I made the change I got the following error.  I presume I should be using something slightly differently?

```
ERROR: LoadError: LoadError: LoadError: LoadError: syntax: incomplete: "function" at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/shallow_water_model.jl:103 requires end
Stacktrace:
 [1] top-level scope at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/shallow_water_model.jl:103
 [2] include(::Function, ::Module, ::String) at ./Base.jl:380
 [3] include at ./Base.jl:368 [inlined]
 [4] include(::String) at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/ShallowWaterModels.jl:1
 [5] top-level scope at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/ShallowWaterModels.jl:14
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include at ./Base.jl:368 [inlined]
 [8] include(::String) at /home/fpoulin/software/Oceananigans.jl/src/Models/Models.jl:1
 [9] top-level scope at /home/fpoulin/software/Oceananigans.jl/src/Models/Models.jl:8
 [10] include(::Function, ::Module, ::String) at ./Base.jl:380
 [11] include at ./Base.jl:368 [inlined]
 [12] include(::String) at /home/fpoulin/software/Oceananigans.jl/src/Oceananigans.jl:1
 [13] top-level scope at /home/fpoulin/software/Oceananigans.jl/src/Oceananigans.jl:143
 [14] include(::Function, ::Module, ::String) at ./Base.jl:380
 [15] include(::Module, ::String) at ./Base.jl:368
 [16] top-level scope at none:2
 [17] eval at ./boot.jl:331 [inlined]
 [18] eval(::Expr) at ./client.jl:467
 [19] top-level scope at ./none:3
in expression starting at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/shallow_water_model.jl:103
in expression starting at /home/fpoulin/software/Oceananigans.jl/src/Models/ShallowWaterModels/ShallowWaterModels.jl:14
in expression starting at /home/fpoulin/software/Oceananigans.jl/src/Models/Models.jl:8
in expression starting at /home/fpoulin/software/Oceananigans.jl/src/Oceananigans.jl:143
```

Reply:
Ah it might just be complaining about the line below `using Oceananigans.Model`:

```julia
 using .ShallowWaterModels: ShallowWaterModel
```

which I think you shouldn't need anymore.

Reply:
Unfortunately, that didn't seem to work.  If I comment out the line you suggested I still get the same error.

```ERROR: LoadError: UndefVarError: ShallowWaterModel not defined```

Reply:
Looks like there's a typo: do you mean to use the name `ShallowWaterModel`, rather than `ShallowWaterModels` (which has an extra "s" on the end?)

Reply:
Ah sorry I think you need to `export ShallowWaterModel` in `src/Models.jl` as well.

Reply:
Sorry @ali-ramadhan that I wasn't able to get this working not now.  I am hoping that @glwagner will approve my changes soon and this can merge.  Maybe you can help me fix this in the next version?  If you think it can wait that is.

Reply:
Sounds good! Certainly it's not a huge issue, excited to see a shallow water model come to life!